strings/base_reference_produce.h

```diff
@@ -261,6 +261,95 @@ WINRT_EXPORT namespace winrt::Windows::Foundation
     }
 }
 
+namespace winrt::impl
+{
+    template <typename T, typename From>
+    T unbox_value_type(From&& value)
+    {
+        static_assert(!is_com_interface_v<T>);
+
+        if (!value)
+        {
+            throw hresult_no_interface();
+        }
+        if constexpr (std::is_enum_v<T>)
+        {
+            if (auto temp = value.try_as<Windows::Foundation::IReference<T>>())
+            {
+                return temp.Value();
+            }
+            else
+            {
+                return static_cast<T>(value.as<Windows::Foundation::IReference<std::underlying_type_t<T>>>().Value());
+            }
+        }
+#ifdef WINRT_IMPL_IUNKNOWN_DEFINED
+        else if constexpr (std::is_same_v<T, GUID>)
```

Comment:
Are shouty-`GUID` and `guid` the same type?

Reply:
Sadly no. Shouty-GUID is the Win32 guid. Lowercase guid is the Windows Runtime Guid.

Reply:
They couldn't be the same type because GUID is defined repeatedly as a macro in different places and that breaks C++ modules.